api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -1056,6 +1066,44 @@ private ValuesLinkingPair getValueSelectsAndJoins(List<DomainValuePair> domainVa
     return new ValuesLinkingPair(valueSelects, valueJoins, domainTable);
   }
 
+  @Override
+  public List<String> getPersonIdsWithWholeGenome(DbDataset dataSet) {
```

Comment:
@freemabd please check this logic and `cohortQueryBuilder.buildUnionedParticipantIdQuery()`

@ericsong please review the rest

Reply:
Does this code need to account for datasets that contain all participants and no cohorts?
![Screen Shot 2021-04-14 at 9 51 36 AM](https://user-images.githubusercontent.com/3904919/114731231-2f5eb580-9d07-11eb-87e9-d886e84fcbe1.png)

Reply:
Good call - thanks. Yes, it does need to. Updated  accordingly, PTAL